src/lib/env-vars.ts

```diff
@@ -109,42 +109,56 @@ export const SUPERVISOR_CONFIG_VAR_PROPERTIES: {
 	},
 };
 
-export const HOST_CONFIG_VAR_PROPERTIES: {
-	[k: string]: JSONSchema6Definition;
-} = {
-	RESIN_HOST_CONFIG_disable_splash: {
-		enum: ['0', '1'],
-		description: 'Enable / Disable the rainbow splash screen',
-		default: '1',
+export const DEVICE_TYPE_SPECIFIC_CONFIG_VAR_PROPERTIES: Array<{
+	capableDeviceTypes: Set<string>;
+	properties: Dictionary<JSONSchema6Definition>;
+}> = [
+	{
+		capableDeviceTypes: new Set([
+			'raspberry-pi',
+			'raspberry-pi2',
+			'raspberrypi3-64',
+			'raspberrypi3',
+			'raspberrypi4-64',
+			'fincm3',
+			'revpi-core-3',
+			'npe-x500-m3',
+		]),
+		properties: {
+			RESIN_HOST_CONFIG_disable_splash: {
+				enum: ['0', '1'],
+				description: 'Enable / Disable the rainbow splash screen',
+				default: '1',
+			},
+			RESIN_HOST_CONFIG_dtparam: {
+				type: 'string',
+				description: 'Define DT parameters',
+				default: '"i2c_arm=on","spi=on","audio=on"',
+			},
+			RESIN_HOST_CONFIG_enable_uart: {
+				enum: ['0', '1'],
+				description: 'Enable / Disable UART',
+				default: '1',
+			},
+			RESIN_HOST_CONFIG_gpu_mem: {
+				type: 'integer',
+				description: 'Define device GPU memory in megabytes.',
+				default: 16,
+			},
+		},
 	},
-	RESIN_HOST_CONFIG_dtparam: {
-		type: 'string',
-		description: 'Define DT parameters',
-		default: '"i2c_arm=on","spi=on","audio=on"',
+	{
+		capableDeviceTypes: new Set(['up-board']),
+		properties: {
+			RESIN_HOST_CONFIGFS_ssdt: {
```

Comment:
Not sure whether this should be
```suggestion
			BALENA_HOST_CONFIGFS_ssdt: {
```
given that this is a brand new config var.
Ended up sticking with RESIN for consistency.